src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs

```diff
@@ -1843,6 +1862,13 @@ internal Pipeline CreatePipeline(RemoteRunspace remoteRunspace)
 
             pipeline.Commands.Clear();
 
+            if (_scriptBlockInitialLocation != null)
+            {
+                var command = new Command("Set-Location");
+                command.Parameters.Add("LiteralPath", _scriptBlockInitialLocation);
```

Comment:
Here can be a security issue. 
Should we check that is really directory and it is impossible inject bad code?

/cc @PaulHigin @TravisEz13

Reply:
This is a really good point.

I do check the directory to see if its valid in `StartJob.cs` [here](https://github.com/davinci26/PowerShell/blob/837eb6a5b6a389a080b5685ea87a27bbb0021169/src/System.Management.Automation/engine/remoting/commands/StartJob.cs#L611) and throw if its not.

The reason I did not make the check in the `PSRemotingCmdlet` is that I wanted to enable the scenario where this parameter is expanded to other cmdlets that inherit from `PSRemotingCmdlet`. 

That being said I could be really defensive and add a command that checks that the string provided by the user is indeed a valid path.

Reply:
Adding a command should be considered like parameterized SQL.  

Impossible is the wrong bar here.  

@iSazonov I'm not sure what the problem is with setting the working directory with a string parameter (this is not a scriptblock).